### PR TITLE
Adds automatic formatting and linting

### PR DESCRIPTION
**Pull request by samouri**
**summary**
This PR adds in automatic code formatting and linting. It also adds make commands as well as a git hook so that these commands are automatically run when performing a `git push`. If you'd like to skip the hook, you can use `git push --no-verify`.

I recommend installing a prettier plugin for whichever IDE/code editor you are using.

new commands:
- `make install`: install the dependencies needed for `format` and `lint`.
- `make lint`: run eslint on all of your javascript files.
- `make format`: run [prettier](https://github.com/prettier/prettier) on the html/css/js files, as well as [google-java-format](https://github.com/google/google-java-format) on the java files.

**next steps**
-  @codyjrivera to run `make format` and `make lint` until everything is passing :). May also need to install node/npm if not already available.

### CODE REVIEW

File: portfolio/Makefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+install:
+	npm install --no-package-lock --no-audit
+	echo "#!/bin/sh\n\tcd portfolio && make format && make lint" > ../.git/hooks/pre-push
+	chmod +x ../.git/hooks/pre-push
+
+format: install
+	npx prettier --write --ignore-path=target --ignore-path=src/main/webapp/node_modules --ignore-path=src/main/webapp/dist src/main/webapp/**/*.{html,css,js}
```
Comment (codyjrivera) [inline portfolio/Makefile:7]: I added extra --ignore-path and --ignore-pattern arguments to prevent extraneous linting errors.

**Comment by samouri** [comment]
@codyjrivera: do you mind if we keep these two changes as two separate PRs:
1. Add in linting rules / autoformatting
2. Format all the code / fix lint errors

**Comment by codyjrivera** [comment]
I cherry-picked my lint fixes out of this PR and put them into another feature branch.

**Comment by samouri** [comment]
LGTM @codyjrivera

**Pull request merged**